src/poseidon/traits.rs

```diff
@@ -0,0 +1,416 @@
+        // - there is no i and j such that x[i] + y[j] = p
+        // - the resultant MDS passes all the three tests
+
+        let xs = lfsr.get_field_elements_mod_p::<Self>(rate + 1);
+        let ys = lfsr.get_field_elements_mod_p::<Self>(rate + 1);
+
+        for i in 0..(rate + 1) {
+            for j in 0..(rate + 1) {
+                mds[i][j] = (xs[i] + &ys[j]).inverse().unwrap();
+            }
+        }
+
+        (ark, mds)
+    }
+}
+
+macro_rules! impl_poseidon_default_parameters_field {
+    ($field: ident, $params: ident) => {
+        impl<P: $params + PoseidonDefaultParameters> PoseidonDefaultParametersField for $field<P> {
+            fn get_default_parameters(
+                rate: usize,
+                optimized_for_constraints: bool,
+            ) -> Option<PoseidonParameters<Self>> {
+                Self::get_default_parameters_internal::<P>(rate, optimized_for_constraints)
+            }
+        }
+    };
+}
+
+impl_poseidon_default_parameters_field!(Fp64, Fp64Parameters);
+impl_poseidon_default_parameters_field!(Fp256, Fp256Parameters);
+impl_poseidon_default_parameters_field!(Fp320, Fp320Parameters);
+impl_poseidon_default_parameters_field!(Fp384, Fp384Parameters);
+impl_poseidon_default_parameters_field!(Fp448, Fp448Parameters);
+impl_poseidon_default_parameters_field!(Fp768, Fp768Parameters);
+impl_poseidon_default_parameters_field!(Fp832, Fp832Parameters);
+
+#[cfg(test)]
+mod test {
+    use crate::poseidon::{PoseidonDefaultParameters, PoseidonDefaultParametersField};
+    use ark_ff::{field_new, fields::Fp256};
+    use ark_ff::{BigInteger256, FftParameters, Fp256Parameters, FpParameters};
+    use ark_test_curves::bls12_381::FrParameters;
+
+    pub struct TestFrParameters;
+
+    impl Fp256Parameters for TestFrParameters {}
+    impl FftParameters for TestFrParameters {
+        type BigInt = <FrParameters as FftParameters>::BigInt;
+        const TWO_ADICITY: u32 = FrParameters::TWO_ADICITY;
+        const TWO_ADIC_ROOT_OF_UNITY: Self::BigInt = FrParameters::TWO_ADIC_ROOT_OF_UNITY;
+    }
+
+    // This TestFrParameters is the same as the BLS12-381's Fr.
+    // MODULUS = 52435875175126190479447740508185965837690552500527637822603658699938581184513
+    impl FpParameters for TestFrParameters {
+        const MODULUS: BigInteger256 = FrParameters::MODULUS;
+        const MODULUS_BITS: u32 = FrParameters::MODULUS_BITS;
+        const CAPACITY: u32 = FrParameters::CAPACITY;
+        const REPR_SHAVE_BITS: u32 = FrParameters::REPR_SHAVE_BITS;
+        const R: BigInteger256 = FrParameters::R;
+        const R2: BigInteger256 = FrParameters::R2;
+        const INV: u64 = FrParameters::INV;
+        const GENERATOR: BigInteger256 = FrParameters::GENERATOR;
+        const MODULUS_MINUS_ONE_DIV_TWO: BigInteger256 = FrParameters::MODULUS_MINUS_ONE_DIV_TWO;
+        const T: BigInteger256 = FrParameters::T;
+        const T_MINUS_ONE_DIV_TWO: BigInteger256 = FrParameters::T_MINUS_ONE_DIV_TWO;
+    }
```

Comment:
Why not just set `TestFrParameters` to be `bls12_381::FrParameters`?

Reply:
Since this TestFrParameters would be slightly different in that it has:
`impl PoseidonDefaultParameters for TestFrParameters`

and I cannot add this to the existing struct `bls12_381::FrParameters` that is in another crate (as this is the test mod).